### PR TITLE
server.js unit test support

### PR DESCRIPTION
**Pull request by abildw**
Wasn't entirely sure how to test the "Server.create" function if anyone has a suggestion.

@geoff-reid @brianparry @srinia6 @dalebremner @tldavies

### CODE REVIEW

File: spec/lib/server-spec.js [hunk 1]
```diff
@@ -4,7 +4,141 @@
...
 
 describe("Server", function() {
+    var server;
+    var parser;
+    var core;
+    var packetDHCP;
+    var outPort =
+        {
+            LegacyPort: (68),
+                EFIPort:(4011)
```
Comment (benbp) [inline spec/lib/server-spec.js]: Indentation.

File: spec/lib/server-spec.js [hunk 2]
```diff
@@ -4,7 +4,141 @@
...
+        };
+
+    before('Server before', function() {
+        helper.setupInjector(
+            [
+            helper.require('/lib/server'),
```
Comment (benbp) [inline spec/lib/server-spec.js]: Indentation.

File: spec/lib/server-spec.js [hunk 3]
```diff
@@ -4,7 +4,141 @@
...
+    });
+
+    describe('Send Function', function() {
+        var fakeData = {
+        };
+        var testServer;
+        before('before Send function',function() {
+            testServer = new server('1',outPort,'123.4.56.7');
+            fakeData = {
+            packetBuffer: 240,
```
Comment (benbp) [inline spec/lib/server-spec.js]: Indentation.

**Comment by benbp** [comment]
I pointed out a couple of places where it would be nice to fix the indentation.

**Comment by benbp** [comment]
Also, I'm not sure how this is possible, but you adding tests has *reduced* coverage...probably a bug with the code coverage tool or something???

**Comment by geoff-reid** [comment]
I added a couple tests to increase the coverage, this should now be ready to go.

**Comment by geoff-reid** [comment]
:+1:  on Billy's tests.

**Comment by srinia6** [comment]
+1

**Comment by jlongever** [comment]
+1

**Pull request merged**